src/_pytest/timing.py

```diff
@@ -20,6 +21,47 @@
     from pytest import MonkeyPatch
 
 
+@dataclasses.dataclass(frozen=True)
+class Instant:
+    """
+    Represents an instant in time, used to both get the timestamp value and to measure
+    the duration of a time span.
+
+    Inspired by Rust's `std::time::Instant`.
+    """
+
+    # Creation time of this instant, using time.time(), to measure actual time.
+    # Use a `lambda` to initialize the default to correctly get the mocked time via `MockTiming`.
+    time: float = dataclasses.field(default_factory=lambda: time(), init=False)
+
+    # Performance counter tick of the instant, used to measure precise elapsed time.
+    # Use a `lambda` to initialize the default to correctly get the mocked time via `MockTiming`.
+    perf_count: float = dataclasses.field(
+        default_factory=lambda: perf_counter(), init=False
+    )
+
+    def duration(self) -> Duration:
```

Comment:
```suggestion
    def until_current(self) -> Duration:
```

Reply:
"current" seems a bit weird to me, how about `until_now`? as in "get me the duration until now"?

Reply:
I agree

Reply:
absolutely, thanks for researching this :+1:

Reply:
While at it, noticed that `Duration.seconds` reads much better than `Duration.elapsed_s`, as in `instant.elapsed().seconds` vs `instant.elapsed().elapsed_s`, so I renamed that as well. 👍